ouroboros-consensus-cardano/tools/db-analyser/Analysis.hs

```diff
@@ -392,148 +426,188 @@ checkNoThunksEvery
+    configLedgerDb = LedgerDbCfg {
+        ledgerDbCfgSecParam = configSecurityParam cfg
+      , ledgerDbCfg         = ExtLedgerCfg cfg
+      }
+
+    f blk oldLedger newLedger = do
+      let traces = HasAnalysis.emitTraces $
+                   HasAnalysis.WithLedgerState blk (ledgerState oldLedger) (ledgerState newLedger)
       mapM_ Debug.traceMarkerIO traces
-      return $ newLedger
 
 {-------------------------------------------------------------------------------
   Auxiliary: processing all blocks in the DB
 -------------------------------------------------------------------------------}
 
-data Limit = Limit Int | Unlimited
-
-decreaseLimit :: Limit -> Maybe Limit
-decreaseLimit Unlimited = Just Unlimited
-decreaseLimit (Limit 0) = Nothing
-decreaseLimit (Limit n) = Just . Limit $ n - 1
-
-data NextStep = Continue | Stop
-
-
-processAllUntil ::
-     forall blk b st. (HasHeader blk, HasAnnTip blk)
-  => Either (ImmutableDB IO blk) (ChainDB IO blk)
-  -> ResourceRegistry IO
-  -> BlockComponent blk b
-  -> ExtLedgerState blk ValuesMK
-  -> Limit
-  -> st
-  -> (st -> b -> IO (NextStep, st))
-  -> IO st
-processAllUntil = either processAllImmutableDB processAllChainDB
-
-processAll ::
-     forall blk b st. (HasHeader blk, HasAnnTip blk)
-  => Either (ImmutableDB IO blk) (ChainDB IO blk)
-  -> ResourceRegistry IO
-  -> BlockComponent blk b
-  -> ExtLedgerState blk ValuesMK
-  -> Limit
-  -> st
-  -> (st -> b -> IO st)
-  -> IO st
-processAll db rr blockComponent initLedger limit initSt cb =
-  processAllUntil db rr blockComponent initLedger limit initSt callback
-    where
-      callback st b = (Continue, ) <$> cb st b
-
-processAll_ ::
-     forall blk b. (HasHeader blk, HasAnnTip blk)
-  => Either (ImmutableDB IO blk) (ChainDB IO blk)
-  -> ResourceRegistry IO
-  -> BlockComponent blk b
-  -> ExtLedgerState blk ValuesMK
-  -> Limit
-  -> (b -> IO ())
-  -> IO ()
-processAll_ db registry blockComponent initLedger limit callback =
-    processAll db registry blockComponent initLedger limit () (const callback)
-
-processAllChainDB ::
-     forall st blk b. (HasHeader blk, HasAnnTip blk)
-  => ChainDB IO blk
-  -> ResourceRegistry IO
-  -> BlockComponent blk b
-  -> ExtLedgerState blk ValuesMK
-  -> Limit
-  -> st
-  -> (st -> b -> IO (NextStep, st))
-  -> IO st
-processAllChainDB chainDB registry blockComponent (ExtLedgerState {..}) limit initState callback = do
-    itr <- case headerStateTip headerState of
-      Origin           -> ChainDB.streamAll
-                             chainDB
-                             registry
-                             blockComponent
-      NotOrigin annTip -> ChainDB.streamFrom
-                             (StreamFromExclusive $ annTipPoint annTip)
-                             chainDB
-                             registry
-                             blockComponent
-    go itr limit initState
+initialLedger ::
```

Comment:
We call this function initial ledger but it gives us back a ledger state. Would it make sense to suffix this with `State`?